### PR TITLE
Fix module uninstallation from the Module Manager

### LINKED ISSUE #25340: BO - Actions (Install - Uninstall - Disable - Enable - Reset) are NOK - You have requested a non-existent service "templating
#### Describe the bug

In the Module Catalog / Module manager page, when we try to Install - Uninstall - Disable - Enable - Reset a module, an error 500 is displayed and an alert "Could not perform action xxx for module undefined" is displayed

#### Steps to Reproduce

1. Go to BO > Modules > Module Catalog > Try to install a module
2. See error 
image
3. Go to BO > Modules > Module Manager page 
4. search the module just installed => well installed
5. Try Uninstall - Disable - Enable - Reset any module
6. See error

#### Additional information

* PrestaShop version: develop
* PHP version: 7.2

### LINKED ISSUE #25441: Cannot uninstall module from the Module Manager
#### Describe the bug

Try to uninstall any module from the module manager, you'll see the error saying "Cannot perform the action on the undefined module"

#### Expected behavior

Successful uninstallation of the module

#### Steps to Reproduce

1. Open the Module manager
2. Try to uninstall a module

#### Additional information

* PrestaShop version: 8.0.0
* PHP version: N/A

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | All details in the related issue
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25340
| How to test?      | All details in the related issue
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.
| Sponsor company | impSolutions

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25442)

### CODE REVIEW

**Review by matks — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by sarahdib** [comment]
Hello @kpodemski 

I noticed that I can not disable/enable/re-install a module. 
Do you think it's possible to fix the other action in this PR too ?

Thank you

**Comment by kpodemski** [comment]
Hello @sarahdib 

Fixed. This means that this PR fixes this issue too: #25340 

@atomiix take a look if it's ok, I guess it is as it's recommended by Symfony to use `twig` directly, right?

**Comment by sarahdib** [comment]
@kpodemski Thank you 😊 

I found a regression :  https://drive.google.com/file/d/1-MqjD8k44nhguh_KuOHriR1mP9MxtNE5/view
1. In module catalog, install a module the first time the "configure" button does not appear and a message indicates that the module is install
2. Click again on install button, the button "configure" appear and a message indicates that the module is install

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]
LGTM! Thanks @kpodemski

**Comment by kpodemski** [comment]
@sarahdib I think it's not related to this PR but module manager needs some additional tests after migration to TypeScript, that's for sure

**Comment by sarahdib** [comment]
@kpodemski 

Without the pr : https://drive.google.com/file/d/1mRf4NDAmBnKFzFjbDo0SspKgHa1d8MkN/view
With the pr : https://drive.google.com/file/d/1-MqjD8k44nhguh_KuOHriR1mP9MxtNE5/view

Without the pr the installation is broken, with the pr the installation works but the configure button is not display and we need to click a second time on "install" to see the "configure" button
In previous version "configure" appear directly after the first click on install button

**Comment by kpodemski** [comment]
Hey @atomiix 

Need a confirmation from you that it's ok, this issue with module data not being refreshed is probably coming from:
https://github.com/PrestaShop/PrestaShop/commit/b958a9ce0839c56d11231d4a29322c4e5fa4397c#diff-48f323312e33dee8b246160e499261db107079773cab85278b3513ca0671a2aa

**Comment by atomiix** [comment]
@kpodemski I'm not sure, I fixed something similar before sf 4.4 was merged in develop: https://github.com/PrestaShop/PrestaShop/pull/24286

**Comment by kpodemski** [comment]
@atomiix IMO this is different because `getModule` in `ModuleRepository` without my change returns `installed` as `0` on the first request after the installation

**Comment by matks** [comment]
Hello guys I'm a bit lost :) can you explain the status of this PR?

**Comment by kpodemski** [comment]
@matks I just rebased it to fix failed sanity tests, let's hope for green CI and then I think QA can take it after reviews

I still think that @atomiix needs a bit of time with this one tho :D

File: src/Core/Addon/Module/ModuleRepository.php [hunk 1]
```diff
@@ -422,9 +424,9 @@ private function getAddonsCatalogModules()
      *
      * @return Module
      */
-    public function getModule($name, $skip_main_class_attributes = false)
+    public function getModule($name, $skip_main_class_attributes = false, $force = false)
```
Comment (Progi1984) [inline src/Core/Addon/Module/ModuleRepository.php]:
```suggestion
    public function getModule($name, $skip_main_class_attributes = false, bool $force = false)
```

Could you add the missing PHPDoc, too ?

**Review by Progi1984 — COMMENTED** [review]
A small comment

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @kpodemski 

Thanks for your PR :rocket: 
LGTM, QA :heavy_check_mark: 
https://drive.google.com/file/d/13iATU4Tnk_KAAJ1gU1yHf41vSZGhqQ-0/view

PS: the only concern that I have is when I click on **Configure** buton of the **1-click Upgrade** module I get error 500, see the attached screen record below:
https://drive.google.com/file/d/1KJ5VAAiwv1ThniOlLR4Jmp4JR3WyqG89/view
This happen only with this module, I didn't encounter any issue with other modules.

Waiting for your feedback.
Thanks!

**Comment by kpodemski** [comment]
Thank you @hibatallahAouadni 

It might be some issue with your installation. It's working for me:
https://www.dropbox.com/s/urze5fvn15vzen3/Zrzut%20ekranu%202021-08-16%20o%2011.25.09.png?dl=0

**Comment by hibatallahAouadni** [comment]
We may need someone else to test the PR cause even when I redo everything from the begining, I still have the same error when I click on **Configure** :disappointed:

**Comment by SD1982** [comment]
Hi @kpodemski  and @hibatallahAouadni 
I have also tested this PR and i confirm that there is an issue with the autoupgrade module 
After installation I have this when I try to configure the autoupgrade module 
image
As we cannot install the module without your PR we cannot determine if the problem is caused by the PR so I suggest that we validate the PR and that we create another issue for the problem of the autoupgrade module 
@Robin-Fischer-PS WDYT ?
Except that everything seems working fine :+1:

**Comment by Robin-Fischer-PS** [comment]
OK for me for QA :heavy_check_mark: on this PR and creating an issue for the autoupgrade module issue :)

**Comment by PierreRambaud** [comment]
Thank you @kpodemski

**Pull request merged**